src/NosCore.GameObject/Networking/ClientSession/ClientSession.cs

```diff
@@ -368,19 +371,19 @@ public void ChangeMapInstance(Guid mapInstanceId, int? mapX, int? mapY)
             }
         }
 
-        private void LeaveMap()
+        private async Task LeaveMap()
         {
-            SendPacket(new MapOutPacket());
-            Character.MapInstance.SendPacket(Character.GenerateOut(),
-                new EveryoneBut(Channel.Id));
+            await SendPacket(new MapOutPacket());
+            await Character.MapInstance!.SendPacket(Character.GenerateOut(),
+                new EveryoneBut(Channel!.Id));
         }
 
         public string GetMessageFromKey(LanguageKey languageKey)
         {
             return Language.Instance.GetMessageFromKey(languageKey, Account.Language);
         }
 
-        public async Task HandlePackets(IEnumerable<IPacket> packetConcatenated, IChannelHandlerContext contex = null)
+        public async Task HandlePackets(IEnumerable<IPacket> packetConcatenated, IChannelHandlerContext? contex = null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5262615)